pkg/resources/password_policy.go

```diff
@@ -0,0 +1,447 @@
+		Default:      1,
+		Description:  "Specifies the minimum number of lowercase characters the password must contain. Supported range: 0 to 256, inclusive. Default: 1",
+		ValidateFunc: validation.IntBetween(0, 256),
+	},
+	"min_numeric_chars": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      1,
+		Description:  "Specifies the minimum number of numeric characters the password must contain. Supported range: 0 to 256, inclusive. Default: 1",
+		ValidateFunc: validation.IntBetween(0, 256),
+	},
+	"min_special_chars": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      1,
+		Description:  "Specifies the minimum number of special characters the password must contain. Supported range: 0 to 256, inclusive. Default: 1",
+		ValidateFunc: validation.IntBetween(0, 256),
+	},
+	"max_age_days": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      90,
+		Description:  "Specifies the maximum number of days before the password must be changed. Supported range: 0 to 999, inclusive. A value of zero (i.e. 0) indicates that the password does not need to be changed. Snowflake does not recommend choosing this value for a default account-level password policy or for any user-level policy. Instead, choose a value that meets your internal security guidelines. Default: 90, which means the password must be changed every 90 days.",
+		ValidateFunc: validation.IntBetween(0, 999),
+	},
+	"max_retries": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      5,
+		Description:  "Specifies the maximum number of attempts to enter a password before being locked out. Supported range: 1 to 10, inclusive. Default: 5",
+		ValidateFunc: validation.IntBetween(1, 10),
+	},
+	"lockout_time_mins": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      15,
+		Description:  "Specifies the number of minutes the user account will be locked after exhausting the designated number of password retries (i.e. PASSWORD_MAX_RETRIES). Supported range: 1 to 999, inclusive. Default: 15",
+		ValidateFunc: validation.IntBetween(1, 999),
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Adds a comment or overwrites an existing comment for the password policy.",
+	},
+}
+
+func PasswordPolicy() *schema.Resource {
+	return &schema.Resource{
+		Description: "A password policy specifies the requirements that must be met to create and reset a password to authenticate to Snowflake.",
+		Create:      CreatePasswordPolicy,
+		Read:        ReadPasswordPolicy,
+		Update:      UpdatePasswordPolicy,
+		Delete:      DeletePasswordPolicy,
+
+		Schema: passwordPolicySchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreatePasswordPolicy implements schema.CreateFunc.
+func CreatePasswordPolicy(d *schema.ResourceData, meta interface{}) error {
+	manager, err := snowflake.NewPasswordPolicyManager()
```

Comment:
the word "manager" is not very descriptiive or useful. What i was thinking is that we could have a Client struct that has references to each object API. e.g.

```
type Client struct {
	conn *sql.DB

	Users           Users
	Roles           Roles
	Warehouses      Warehouses
	Databases       Databases
	Schemas         Schemas
	Tables          Tables
	NetworkPolicies NetworkPolicies
}
```
where Roles for example would be:

```
// Roles describes all the roles related methods that the
// Snowflake API supports.
type Roles interface {
	// List all the roles by pattern.
	List(ctx context.Context, options RoleListOptions) ([]*Role, error)
	// Create a new role with the given options.
	Create(ctx context.Context, options RoleCreateOptions) (*Role, error)
	// Read an role by its name.
	Read(ctx context.Context, role string) (*Role, error)
	// Update attributes of an existing role.
	Update(ctx context.Context, role string, options RoleUpdateOptions) (*Role, error)
	// Delete a role by its name.
	Delete(ctx context.Context, role string) error
	// Rename a role name.
	Rename(ctx context.Context, old string, new string) error
}

```
then you could reference password policeis as: `client.PasswordPolicies.Create(<CreatePasswordPolicyInput>)`

I also think addtion a validation method to the input would be useful. Example:

```
// RoleCreateOptions represents the options for creating a role.
type RoleCreateOptions struct {
	*RoleProperties

	// Required: Identifier for the role; must be unique for your account.
	Name string
}

func (o RoleCreateOptions) validate() error {
	if o.Name == "" {
		return errors.New("name must not be empty")
	}
	return nil
}
```
then you could validate input before doing generic request as so:

```
func (r *roles) List(ctx context.Context, options RoleListOptions) ([]*Role, error) {
	if err := options.validate(); err != nil {
		return nil, fmt.Errorf("validate list options: %w", err)
	}
...
```